internal/cfgplugins/interface.go

```diff
@@ -756,3 +791,102 @@ func SetupStaticAggregateAtomically(t *testing.T, dut *ondatra.DUTDevice, aggrBa
 	}
 	return agg
 }
+
+// AddPortToAggregate adds an Ondatra port as a member to the aggregate interface.
+func AddPortToAggregate(t *testing.T, dut *ondatra.DUTDevice, aggID string, dutAggPorts []*ondatra.Port, b *gnmi.SetBatch, op *ondatra.Port) {
+	gnmi.BatchDelete(b, gnmi.OC().Interface(op.Name()).Ethernet().AggregateId().Config())
+
+	d := &oc.Root{}
+	i := d.GetOrCreateInterface(op.Name())
+	i.Description = ygot.String("LAG - Member - " + op.Name())
+	e := i.GetOrCreateEthernet()
+	e.AggregateId = ygot.String(aggID)
+	i.Type = oc.IETFInterfaces_InterfaceType_ethernetCsmacd
+
+	if deviations.InterfaceEnabled(dut) {
+		i.Enabled = ygot.Bool(true)
+	}
+	if op.PMD() == ondatra.PMD100GBASEFR && deviations.ExplicitPortSpeed(dut) {
+		e.AutoNegotiate = ygot.Bool(false)
+		e.DuplexMode = oc.Ethernet_DuplexMode_FULL
+		e.PortSpeed = oc.IfEthernet_ETHERNET_SPEED_SPEED_100GB
+	}
+	gnmi.BatchReplace(b, gnmi.OC().Interface(op.Name()).Config(), i)
+}
+
+// AddSubInterface adds a subinterface to an interface.
+func AddSubInterface(t *testing.T, dut *ondatra.DUTDevice, b *gnmi.SetBatch, i *oc.Interface, s *DUTSubInterfaceData) {
+	sub := i.GetOrCreateSubinterface(uint32(s.VlanID))
+	sub.Enabled = ygot.Bool(true)
+	if s.VlanID != 0 {
+		sub.GetOrCreateVlan().GetOrCreateMatch().GetOrCreateSingleTagged().VlanId = ygot.Uint16(uint16(s.VlanID))
+	}
+	if s.IPv4Address == nil && s.IPv6Address == nil {
+		t.Fatalf("No IPv4 or IPv6 address found for  %s or a subinterface under this lag", i.GetName())
+	}
+	if s.IPv4Address != nil {
+		sub.GetOrCreateIpv4().GetOrCreateAddress(s.IPv4Address.String()).PrefixLength = ygot.Uint8(uint8(s.IPv4PrefixLen))
+		if deviations.IPv4MissingEnabled(dut) {
+			sub.GetOrCreateIpv4().SetEnabled(true)
+		}
+	}
+	if s.IPv6Address != nil {
+		sub.GetOrCreateIpv6().GetOrCreateAddress(s.IPv6Address.String()).PrefixLength = ygot.Uint8(uint8(s.IPv6PrefixLen))
+		if deviations.IPv4MissingEnabled(dut) {
+			sub.GetOrCreateIpv6().SetEnabled(true)
+		}
+	}
+	gnmi.BatchReplace(b, gnmi.OC().Interface(i.GetName()).Subinterface(uint32(s.VlanID)).Config(), sub)
+}
+
+// NewAggregateInterface creates the below configuration for the aggregate interface:
+// 1. Create a new aggregate interface
+// 2. LACP configuration
+// 3. Adds member Ports configuration to an aggregate interface
+// 4. Subinterface configuration including thier IP address and VLAN ID
+// Note that you will still need to push the batch config to the DUT in your code.
+func NewAggregateInterface(t *testing.T, dut *ondatra.DUTDevice, b *gnmi.SetBatch, l *DUTAggData, aggID string) *oc.Interface {
```

Comment:
I think aggID is not needed?     Use DUTAggData.LagName instead?

```suggestion
func NewAggregateInterface(t *testing.T, dut *ondatra.DUTDevice, b *gnmi.SetBatch, l *DUTAggData) *oc.Interface {
```